stew/objects.nim

```diff
@@ -67,41 +68,34 @@ proc baseType*(obj: RootObj): cstring =
 proc baseType*(obj: ref RootObj): cstring =
   obj[].baseType
 
-when false:
-  # TODO: Implementing this doesn't seem possible at the moment.
-  #
-  # When given enum like:
-  #
-  # type WithoutHoles2 = enum
-  #   A2 = 2, B2 = 3, C2 = 4
-  #
-  # ...the code below will print:
-  #
-  #  EnumTy
-  #    Empty
-  #    Sym "A2"
-  #    Sym "B2"
-  #    Sym "C2"
-  #
-  macro hasHoles*(T: type[enum]): bool =
-    let t = getType(T)[1]
-    echo t.treeRepr
-    return newLit(true)
+macro enumRangeOrd*(a: type[enum]): untyped =
+  ## This macro returns an array with all the ordinal values of an enum
+  let
+    values = a.getType[1][1..^1]
+    valuesOrded = values.mapIt(newCall("ord", it))
+  newNimNode(nnkBracket).add(valuesOrded)
+
+proc contains*(e: type[enum], v: SomeInteger): bool =
+  v in enumRangeOrd(e)
```

Comment:
You could optimize for non-holes enum a bit here:
```nim
when e.hasHoles:
  v in enumRangeOrd(e)
else:
  v in e.low ..< e.high
```

Reply:
Done it, but the _high_ must be included though.